### ISSUE TITLE
Refactoring Entities

### ISSUE THREAD

**Issue by mlaskus** [ojakovlj/Asteroid-Hunter#3]
Entity interface should be an abstract class rather than an interface, as Entities mostly share implementations of methods.

Consider an interface for clickable objects

``` java
public interface Clickable {
    boolean onDown();
    boolean onMove();
    boolean onUp();
}
```

Clickable objects touched with an event, can be iterated over according to the inverse of their draw order. If the raised event returns true, the event is consumed and propagation to objects below stops.

**Comment by ojakovlj** [comment]
Good point on the abstract vs interface, however, is the interface for clickable objects suitable for the situation where multiple variables are exchanged between the individual methods?

**Comment by mlaskus** [comment]
What variables do you need to exchange?

We basically have two separate contexts, the clicked object and the View on which the touch listener was invoked. If you need to remember something between onDown and onUp on a single Entity, then that can be done with properties of that Entity. The same applies for the View.

**Comment by ojakovlj** [comment]
Yes, you are right, it could be done this way. Also, we still need to fix the zoom limiter and implement the instance saving (to prevent it all from resetting on screen orientation change)

**Comment by mlaskus** [comment]
Oh sure, could you open up separate issues for these problems? Then once we have a bit more time it will be easy to see what needs doing.
